src/Gui/Viewer/Viewer.cpp

```diff
@@ -517,8 +517,8 @@ bool Viewer::initializeGL() {
 }
 
 void Viewer::resizeGL( QResizeEvent* event ) {
-    int width  = event->size().width();
-    int height = event->size().height();
+    int width  = event->size().width() * devicePixelRatio();
+    int height = event->size().height() * devicePixelRatio();
```

Comment:
Thanks, I'll give a try. I remember having tried something like this at the beginning of Radium dev but with some difficulties to take devicePixelRatio into account for picking and other pixel access.

Reply:
I've not tested picking nor anything else, but definitively need to be tested before merge.

Reply:
Tested on windows. If this solve the viewport problem, as assumed, there are problems when picking :(. BTW, my suggestions here seems to work (at least for picking). I've tested it on windows and this solves my problem. My suggestion does not impact macos but needs to be tested on Linux.

Reply:
Suggestions tested on Linux/gnome, with hdpi display. Due to the way scaling seems to be managed by gnome, the suggestions do not impact Linux ( `devicePixelRatio()` seems to always return 1).

So, the solution is indeed simple, thanks @dlyr, but should be placed where I suggest (not in the computation of width and height but in the viewport definition). Once done, the I'll approve the PR.

Reply:
On my setup,  devicePixelRatio() returns 1 for 100% and 2 of 200% following gnome display setup. 
Changing only viewport make the rendering to be done as lower resolution, is this what we want ?
Changing viewport only
100%
![view-100](https://user-images.githubusercontent.com/11328509/217187283-0f7d1a0b-6a40-4a71-b7d6-28953c2f263c.png)
200%
![view-200](https://user-images.githubusercontent.com/11328509/217187280-9a0fe6fb-1481-467b-9d0b-a509eb098b5a.png)
Changing dim
100%
![dim-100](https://user-images.githubusercontent.com/11328509/217187278-cdbdffd4-d233-48db-b2e1-4eb038b98aae.png)
200%
![dim-200](https://user-images.githubusercontent.com/11328509/217187274-47530f8a-32e5-49b4-9903-3283ed9c03cf.png)

```{.bash}
$lsb_release -a
Description:	Debian GNU/Linux 11 (bullseye)
$ qmake6 -v
Using Qt version 6.3.1 in /usr/lib/x86_64-linux-gnu
```

Reply:
on my setting, `devicePixelRatio()` always return 1
![scale100](https://user-images.githubusercontent.com/11328578/217239873-ebd99dd4-f313-41b5-83e1-b50980a6abf6.png)
![scale200](https://user-images.githubusercontent.com/11328578/217239900-9d4eccc7-bf3e-4065-98f6-6395de21bf2b.png)
```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 11 (bullseye)
Release:	11
Codename:	bullseye
$ qmake -v
QMake version 3.1
Using Qt version 5.15.2 in /usr/lib/x86_64-linux-gnu
```

If you execute the demo `Picking`, do you have the correct result with your 200% scale setting ? On windows (and Qt6.3.1), I haven't when changing dim, but it was OK when changing viewport (scaling was set at 150% as recommended by windows - and accepted by my eyes ^^ - and I didn't notice quality degradation).
I'll try to install Qt6 on linux to test with the same version than yours.

Reply:
After upgrading Qt version :
```
$ qmake6 --version
QMake version 3.1
Using Qt version 6.4.2 in /usr/lib/x86_64-linux-gnu
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 11 (bullseye)
Release:	11
Codename:	bullseye
```

at 100% scale : `devicePixelRatio()` returns 1.25 (in tweakui, I've set the font to 125% to be able to read my screen)
original (without taking `devicePixelRatio` into account) - picking NOT OK: 
![qt6-scale100-nopixelratio](https://user-images.githubusercontent.com/11328578/217254401-a14e4065-894a-4eb3-9df8-e660407bc521.png)

viewport scaled by devicePixelRatio - picking OK
![qt6-scale100-viewportscaled](https://user-images.githubusercontent.com/11328578/217254987-9ea85094-192a-4bd4-8496-dd584b094b18.png)

at 200% scale : `devicePixelRatio()` returns 2.5
viewport scaled by devicePixelRatio - picking OK - quality degradation
![qt6-scale200-viewportscaled](https://user-images.githubusercontent.com/11328578/217255407-0a009266-7dc1-490d-8eb7-549917f760b5.png)

dim scaled by devicePixelRatio - picking NOT OK - quality OK
![qt6-scale200-dimscaled](https://user-images.githubusercontent.com/11328578/217255585-e9d3d1d8-5341-4eb3-9f7b-332d8d0631fb.png)

So, either we change all the picking management code to take pixelRatio into account, either we just scale the viewport, with a quality degradation acceptable for small scale (e.g 125%)

BTW, changing the dim to prevent quality lost (and changing the way picking is managed), seems strange to me as, e.g., for a scaling at 125%, we request a picture with size 800x600 and get one with size 1000x750 ... could be strange for some apps ...